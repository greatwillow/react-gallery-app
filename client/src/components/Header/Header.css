.header-container {
  position: fixed;
  top: 0px;
  left: 0px;
  right: 0px;
  height: 81px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  padding: 0px 30px 0px 30px;
  z-index: 5;
  background-color: #fff;
  box-shadow: 0 10px 20px rgba(0, 0, 0, 0.19), 0 6px 6px rgba(0, 0, 0, 0.23);
}
