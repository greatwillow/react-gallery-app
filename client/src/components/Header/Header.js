import React, { Component } from 'react';
// COMPONENTS
import SearchBar from '../SearchBar/SearchBar';
import HeaderTitle from '../HeaderTitle/HeaderTitle';

import './Header.css';

class Header extends Component {
  render() {
    return (
      <div className="header-container">
        <HeaderTitle {...this.props} />
        <SearchBar {...this.props} />
      </div>
    );
  }
}

export default Header;
