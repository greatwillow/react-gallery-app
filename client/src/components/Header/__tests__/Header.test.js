import React from 'react';
import Header from '../Header';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('Header component renders the header correctly', () => {
  const props = {
    handleSubmitSearchTerm: jest.fn(),
    handleSearchTermChange: jest.fn(),
    searchTerm: 'test',
    windowWidth: 800
  };

  const header = shallow(<Header {...props} />);

  it('renders correctly', () => {
    expect(header).toMatchSnapshot();
  });
});
