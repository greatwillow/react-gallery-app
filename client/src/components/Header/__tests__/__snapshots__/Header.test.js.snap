// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders the header correctly renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    handleSearchTermChange={[Function]}
    handleSubmitSearchTerm={[Function]}
    searchTerm="test"
    windowWidth={800}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <HeaderTitle
          handleSearchTermChange={[Function]}
          handleSubmitSearchTerm={[Function]}
          searchTerm="test"
          windowWidth={800}
/>,
        <SearchBar
          handleSearchTermChange={[Function]}
          handleSubmitSearchTerm={[Function]}
          searchTerm="test"
          windowWidth={800}
/>,
      ],
      "className": "header-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleSearchTermChange": [Function],
          "handleSubmitSearchTerm": [Function],
          "searchTerm": "test",
          "windowWidth": 800,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleSearchTermChange": [Function],
          "handleSubmitSearchTerm": [Function],
          "searchTerm": "test",
          "windowWidth": 800,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <HeaderTitle
            handleSearchTermChange={[Function]}
            handleSubmitSearchTerm={[Function]}
            searchTerm="test"
            windowWidth={800}
/>,
          <SearchBar
            handleSearchTermChange={[Function]}
            handleSubmitSearchTerm={[Function]}
            searchTerm="test"
            windowWidth={800}
/>,
        ],
        "className": "header-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleSearchTermChange": [Function],
            "handleSubmitSearchTerm": [Function],
            "searchTerm": "test",
            "windowWidth": 800,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleSearchTermChange": [Function],
            "handleSubmitSearchTerm": [Function],
            "searchTerm": "test",
            "windowWidth": 800,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
